File: edge_health_phone/app/src/main/java/com/example/edge_health/SensorDao.java
package com.example.edge_health;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Query;
import androidx.room.Upsert;

@Dao
public interface SensorDao {
    @Upsert
    void insert(SensorsCollect sensorsCollect);

    @Delete
    void delete(SensorsCollect sensorsCollect);

    @Query("DELETE FROM SensorsCollect")
    void deleteAll();

    @Query("SELECT * FROM SensorsCollect")
    SensorsCollect[] getAll();

    @Query("SELECT * FROM SensorsCollect WHERE sent = 0 ORDER BY timestamp ASC LIMIT 20 ")
    SensorsCollect[] getFirst20NotSent();
}

File: edge_health_wear/app/src/main/java/com/example/edge_health_wear/SensorDao.kt
package com.example.edge_health_wear

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Query
import androidx.room.Upsert

@Dao
interface SensorDao {
    @Upsert // Upsert is a combination of insert and update
    suspend fun insertSensorData(sensorData: SensorCollect)

    @Delete
    suspend fun deleteSensorData(sensorData: SensorCollect)

    @Query("DELETE FROM SensorCollect")
    suspend fun deleteAllSensorData()

    @Query("SELECT * FROM SensorCollect ORDER BY timestamp ASC")
    suspend fun getSensorsOrderedByTimestamp(): List<SensorCollect> // Use a LiveData object to observe changes in the database
}